Fleet Desktop device user page

RachelElysia opened the pull request:
Cerra #4092 

TODO:
- [x] High fidelity user page
- [x] Reorganize directory
- [x] Reorganize route/app
- [x] Refactor on new API routes instead of /host route
- [x] Fix auth token bug
- [x] Build out loading state, error states, empty states

SEPARATE PR:
- Build out E2e test with static data

<img width="1552" alt="Screen Shot 2022-03-15 at 5 04 43 PM" src="https://user-images.githubusercontent.com/71795832/158472387-d489e780-8557-4797-a1e6-7d0c936ec325.png">
<img width="1545" alt="Screen Shot 2022-03-15 at 5 04 35 PM" src="https://user-images.githubusercontent.com/71795832/158472391-99840788-841e-4ca1-b62c-6140ba82f4ca.png">
<img width="1549" alt="Screen Shot 2022-03-15 at 5 06 02 PM" src="https://user-images.githubusercontent.com/71795832/158472540-3feb3284-c68c-4b03-9611-809b47d1cee2.png">
<img width="1413" alt="Screen Shot 2022-03-17 at 1 25 41 PM" src="https://user-images.githubusercontent.com/71795832/158859727-cb7f859d-973f-4e7e-8d97-33148a2a1da4.png">


# Checklist for submitter

If some of the following don't apply, delete the relevant line.

- [x] Changes file added (for user-visible changes)
- [ ] Documented any API changes (docs/Using-Fleet/REST-API.md)
- [ ] Added/updated tests
- [x] Manual QA for all new/changed functionality

lukeheath commented:
@RachelElysia Reminder to take PR out of the draft and tag issue #4589

RachelElysia commented:
@lukeheath Thanks! #4092

lukeheath reviewed (changes requested):
✨  This looks great. Awesome job! 

I think there may be opportunity to abstract some of the duplicate software table code to the global context.

RachelElysia commented:
That just saved 400 lines of code! I moved the DeviceUserPage directory into the host directory and moved the SoftwareTab into that directory so it's used by both DeviceUserPage and HostDetailsPage!

Up for re-review!

lukeheath reviewed (approved):
✨  Looks great! Nice code reduction! 🚀
